### PR TITLE
Vue: Fix format on self-closing and empty blocks

### PR DESCRIPTION
**Pull request by fisker**
https://github.com/prettier/prettier/pull/9052#issuecomment-680083242

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
DO you guys think this is bug?

**Prettier 2.1.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeMBXAzjCBbAAgBsBDKAcwF4AdEPEgJwGsATCAdyloD5UB6TDnzcQAGhAQADjACW0LMlCMGHAAqMEClCSLsSATwXiARgxJgmcGAGVJ5mRWQwGGOOLh5jcFi28AZMnIMEnI4ADEIBnoYWUdtDFwxEAALGDwiAHVkmXgsOzA4a00cmQA3HP1kcCwjEAcsOAYYVTNyemQAMx0G8QArLAAPACEzCytrEjw4Pwc4Tu63EH6B6wdyIjgARQwIeHmiHpA7BgaGKtLXJMkGBxgMmRYYZOQADgAGcWuIBoyzSSrrnBTqU5uIAI47eAtKRaEAkLAAWigcG83iSDDgEJkGJaIXaSC6B0WDTwMicLmJaw2212cwJC3EMBIxnuj2eSAATIyzDIiGsAML4fEgIEAViS2DgABVmVpCYcLnAAJJQXywaxgG7SACCqusMH0G32DQAviagA)
```sh
--parser vue
```

**Input:**
```vue
<custom lang="markdown"></custom>
```

**Output:**
```vue
<custom lang="markdown">

</custom>

```

**Comment by thorn0** [comment]
Yes, looks unexpected.

**Comment by sosukesuzuki** [comment]
I agree with thorn0

File: tests/vue/multiparser/jsfmt.spec.js [hunk 1]
```diff
@@ -1 +1,18 @@
...
+    name: "spaces",
+    code: '<custom lang="markdown">   </custom>'
+  },
+  {
+    name: "new line",
+    code: '<custom lang="markdown">\n \n</custom>'
+  },
+  {
+    name: "non-space",
+    code: '<custom lang="markdown">\n \u2005 \n</custom>'
```
Comment (fisker) [inline tests/vue/multiparser/jsfmt.spec.js:16]: This one, html think it's not empty, and `markdown` will print to a empty `Doc`, how do I tell it's empty?

Comment (thorn0) [reply tests/vue/multiparser/jsfmt.spec.js:16]: The only child node is text consisting of whitespace only.

Comment (fisker) [reply tests/vue/multiparser/jsfmt.spec.js]:
```json
{
      "type": "concat",
      "parts": [
        {
          "type": "concat",
          "parts": [
            {
              "type": "fill",
              "parts": []
            }
          ]
        }
      ]
    }
```

Comment (thorn0) [reply tests/vue/multiparser/jsfmt.spec.js]: Try using `normalizeParts` from `doc-utils`. You might want to improve it to handle an empty `fill` doc.

Comment (fisker) [reply tests/vue/multiparser/jsfmt.spec.js]: What if the embed parser return `ifBreak('foo', '')`, we should not be able to know it's empty or not. Am I right?

Comment (thorn0) [reply tests/vue/multiparser/jsfmt.spec.js]: Can't we first check if the content of the tag is empty or whitespace-only and skip the embedded parser if it is?

Comment (fisker) [reply tests/vue/multiparser/jsfmt.spec.js]: I tried, and it works, just feel something not right.

Comment (thorn0) [reply tests/vue/multiparser/jsfmt.spec.js]: It's fine to assume that reformatting an empty document should result in an empty document for any language.

**Comment by fisker** [comment]
Done

**Review by thorn0 — APPROVED** [review]

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**